### ISSUE TITLE
[rllib] InvalidArgumentError with Discrete action space

### ISSUE THREAD

**Issue by pschafhalter** [ray-project/ray#2468]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **Ray installed from (source or binary)**: pip
- **Ray version**: 0.5.0
- **Python version**: 3.5

### Describe the problem

I'm trying to use RLlib to train an agent using a custom model and a custom preprocessor. The environment we're training in has a `Discrete(14)` action space, however training crashes because due to an invalid label. Somehow a label of 14 is generated for the action when the valid range of labels is [0. 14).

### Source code / logs
[train.py](https://github.com/ray-project/ray/files/2225401/train.txt)
[trace.log](https://github.com/ray-project/ray/files/2225409/traceback.txt)

cc @efang96

**Comment by richardliaw** [comment]
Can you do the following:
1. Try setting `simple_optimizer = True` for PPO
2. Set 
```
- layer3 = slim.fully_connected(layer2, 14, activation_fn=tf.nn.softmax)
--------------
+ layer3 = slim.fully_connected(layer2, num_outputs, activation_fn=tf.nn.softmax)
```
3. Also, try `print(trainer.local_evaluator.sample())`

**Comment by efang96** [comment]
We tried doing 2 and 3 without any success.

We also tried 1, but version 0.5.0 installed from pip does not have simple_optimizer as entry in the config.

**Comment by pschafhalter** [comment]
I was able to reproduce the same error with (1), (2), and (3) using Ray installed from source.

**Comment by ericl** [comment]
The attached model seems to return 14 outputs instead of `num_outputs` outputs.
